app/models/mixins/supports_feature_mixin.rb

```diff
@@ -68,19 +116,32 @@ def unsupported
   end
 
   class_methods do
-    def unsupported_reason(feature)
-      public_send("supports_#{feature}?") unless unsupported.key?(feature)
-      unsupported[feature]
-    end
-
+    # This is the DSL used a class level to define what is supported
     def supports(feature, &block)
+      SupportsFeatureMixin.guard_queryable_feature(feature)
       send(:define_supports_methods, feature, true, &block)
```

Comment:
`define_supports_methods` is private, that's why `send` is needed here.
I could make it public, but I want to create those methods only through the `supports` DSL